sdk-python/omnilogger/logger.py

```diff
@@ -18,48 +18,60 @@ def start_listener(url: str):
     sys.stderr = StreamToLogger(logger, logging.ERROR)
 
 
-def handle_openai_response(data: str, url: str):
-    output = data.split("body='")[1].split("' headers=")[0]
-    output = bytes(output, "utf-8").decode("unicode_escape")
-    output = json.loads(output)
-
-    log = {
-        "input": "This is a new dummy input!",
-        "datetime_utc": datetime.utcfromtimestamp(output["created"]),
-        "output": output["choices"][0]["text"],
-        "total_tokens": output["usage"]["total_tokens"],
-    }
-    send_to_db(url, log)
-
-
 class OpenAIFilter(logging.Filter):
     url: str
+    t_prompt: str
 
     def __init__(self, url):
         self.url = url
+        self.t_prompt = ""
 
     def filter(self, record) -> bool:
         msg = record.getMessage()
 
-        if "OpenAI" in msg or "openai" in msg:
+        if "OpenAI" in msg or "openai" in msg or "text-davinci-003" in msg:
             if "message='Request to OpenAI API'" in msg:
                 return False
 
             if "message='Post details'" in msg:
+                try:
+                    self.handle_openai_request(msg)
+                except IndexError:
+                    record.msg = "Prompt not found in OpenAI request: " + msg
+                    return True
                 return False
 
             if "message='OpenAI API response'" in msg:
                 return False
 
             if "message='API response body'" in msg:
                 try:
-                    handle_openai_response(msg, self.url)
+                    self.handle_openai_response(msg)
                 except json.decoder.JSONDecodeError:
                     record.msg = "JSONDecodeError for OpenAI response: " + msg
                     return True
                 return False
+        record.msg += record.levelname
         return True
 
+    def handle_openai_response(self, data: str):
+        output = data.split("body='")[1].split("' headers=")[0]
+        output = bytes(output, "utf-8").decode("unicode_escape")
+        output = json.loads(output)
+
+        log = {
+            "input": self.t_prompt,
+            "datetime_utc": datetime.utcfromtimestamp(output["created"]),
+            "output": output["choices"][0]["text"],
+            "total_tokens": output["usage"]["total_tokens"],
+        }
+        self.t_prompt = ""
+        send_to_db(self.url, log)
+
+    def handle_openai_request(self, data: str):
```

Comment:
This isn't clear to me what this function actually does

Reply:
I added doc strings for both handlers, is it the best way to do that?
This function extracts the prompt from the console log, but I named it using the same convention as the function above

Reply:
`extract_prompt_from_request`